modules/storages/app/services/storages/file_links/create_service.rb

```diff
@@ -39,6 +39,19 @@ def persist(service_result)
 
   private
 
+  def after_perform(service_result)
+    # This only gets called if service_result is successful
+    container = service_result.result.container
+
+    # If the container isn't journaled, no need to proceed
+    return service_result unless container&.class&.journaled?
+
+    # If journal creation fails, we don't care for now
+    Journals::CreateService.new(container, service_result.result.creator).call
```

Comment:
Please use the `save_journals` method on the container. That will ensure the [`JOURNAL_CREATED` event being sent out](https://github.com/opf/openproject/blob/dev/lib_static/plugins/acts_as_journalized/lib/acts/journalized/save_hooks.rb#L67-L69) which e.g. the sending of notifications (leading to in app notifications and mails) relies on.

Admittedly, it would also make sense to move that notification sending into the `Journals::CreateService` since it should be service's responsibility.

Regardless, it is also shorter to just call `container.save_journal` :)

A nice side effect of this change should be, that the container will receive a touch which will make for a consistent experience for the users, seeing the upload in the journals and seeing that the work package was updated at the same time.

Since I haven't tested the code out myself, please also check whether the sending of notifications and mails works as expected. I would expect this to be the case as the changes in here follow the established patterns but double checking is advisable.

Reply:
After the change notifications are working as expected.